experiments/opencog/pattern_matcher_vqa/pattern_matcher_vqa.py

```diff
@@ -345,35 +420,37 @@ def answerOtherQuestion(self, queryInScheme):
                           '%s records, time: %s microseconds',
                           len(resultsData.out), delta.microseconds)
 
-        def extract_predicate(atoms):
-            for atom in atoms:
-                if atom.type == opencog.atomspace.types.InheritanceLink:
-                    if atom.out[1].name != 'BoundingBox':
-                        predicate_name = atom.out[0].name
-                        return predicate_name
+        results = self.sort_results(resultsData, a_extract_predicate=True)
+        if not results:
+            return None, None, None
+        maxResult = results[0]
+        answer = maxResult.get_answer()
```

Comment:
It seems you don't need local variables you can just write:

    return maxResult.get_answer(), maxResult.get_bounding_box_id(), maxResult.get_expression()